pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -130,17 +128,26 @@ func (d *DependenciesDistributor) NeedLeaderElection() bool {
 
 func (d *DependenciesDistributor) discoverResources(period time.Duration) {
 	wait.Until(func() {
-		for _, gvr := range supportedTypes {
-			if d.InformerManager.IsHandlerExist(gvr, d.EventHandler) {
-				continue
+		newResources := lifted.GetDeletableResources(d.DiscoveryClientSet)
+		for gvr := range newResources {
+			if !d.InformerManager.IsHandlerExist(gvr, d.EventHandler) && !d.SkippedResourceConfig.GVRDisabled(gvr, d.RESTMapper) {
+				klog.Infof("Setup informer for %s", gvr.String())
+				d.InformerManager.ForResource(gvr, d.EventHandler)
 			}
-			klog.Infof("Setup informer for %s", gvr.String())
-			d.InformerManager.ForResource(gvr, d.EventHandler)
 		}
 		d.InformerManager.Start()
 	}, period, d.stopCh)
 }
 
+// Filter out cluster scope resources.
+func (d *DependenciesDistributor) Filter(obj interface{}) bool {
```

Comment:
There are still many inappropriate namespaced resources that are not filtered by the function `Filter`. At least, it should be consistent with the function [EventFilter](https://github.com/karmada-io/karmada/blob/master/pkg/detector/detector.go#L244).

Reply:
I have used a new approach. @whitewindmills